server/src/main/java/org/elasticsearch/cluster/routing/ShardRouting.java

```diff
@@ -485,13 +482,14 @@ public ShardRouting reinitializeReplicaShard() {
      * <code>INITIALIZING</code> or <code>RELOCATING</code>. Any relocation will be
```

Comment:
In `public ShardRouting(ShardId shardId, StreamInput in)` we're serializing/de-serializing the shard size only when the state is `state == ShardRoutingState.UNASSIGNED || state == ShardRoutingState.INITIALIZING` I think that's not consistent with the current behaviour?

Reply:
Good point! I've changed the serialization logic to write `expectedShardSize` for `STARTED` shards starting from `8.5.0`